.github/workflows/deploy.yml

```diff
@@ -0,0 +1,70 @@
+# Build the caller repo's Docker image and deploy it to Elastic Beanstalk.
+name: Deploy
+permissions:
+  id-token: write  # Required to use AWS OIDC auth.
+on:
+  workflow_call:
+    inputs:
+      github_environment_name:
+        type: string
+        required: true
+        description: "The GitHub environment name to reference, e.g. 'qa' or 'prod'"
+      github_environment_url:
+        type: string
+        description: "The GitHub environment URL to reference, e.g. 'https://qa-lms.hypothes.is/'"
+      aws_region:
+        type: string
+        required: true
+        description: "The AWS region to deploy to, e.g. 'us-west-1' or 'ca-central-1'"
+      elasticbeanstalk_application:
+        type: string
+        required: true
+        description: "The Elastic Beanstalk application to deploy to, e.g. 'via' or 'lms'"
+      elasticbeanstalk_environment:
+        type: string
+        required: true
+        description: "The Elastic Beanstalk environment to deploy to, e.g. 'qa' or 'prod'"
+      docker_tag:
+        type: string
+        description: "The Docker tag to deploy, e.g. '20221019-g8a52aea'"
+      type:
+        type: string
+        required: true
+        description: "The type of action to perform: `deploy` to deploy the given `docker_tag`; `redeploy` to re-deploy the current version of the app; or `sync-env` to synchronize the Elastic Beanstalk environment definition"
```

Comment:
I tweaked the style of these input names. Partially to try to make it clearer what's what (what belongs to GitHub, AWS, Docker or Elastic Beanstalk) and partially to use a lower-case style. GitHub's docs about inputs [recommend using lowercase inputs IDs](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs) (although they then immediately give an example with uppercase letters in the input IDs! :shrug:)

| Original Name | New Name                                     |
| ------------- | -------------------------------------------- |
|               | `github_environment_name`                    |
|               | `github_environment_url`                     |
| `Region`      | `aws_region`                                 |
| `Application` | `elasticbeanstalk_application`               |
| `Environment` | `elasticbeanstalk_environment`               |
| `Version`     | `docker_tag                                  |
| `Operation`   | `type` (passed to `init.sh`'s `TYPE` envvar) |

Reply:
I renamed `type` back to `operation` (lower-case): `type` is a terrible name